helix-core/src/main/java/org/apache/helix/task/TaskRunner.java

```diff
@@ -208,7 +209,10 @@ private static boolean setRequestedState(HelixDataAccessor accessor, String inst
         String.format("Requesting a state transition to %s for partition %s.", state, partition));
     try {
       PropertyKey.Builder keyBuilder = accessor.keyBuilder();
-      PropertyKey key = keyBuilder.currentState(instance, sessionId, resource);
+      PropertyKey key =
+          Boolean.getBoolean(SystemPropertyKeys.TASK_CURRENT_STATE_PATH_DISABLED) ? keyBuilder
```

Comment:
Let's have the path build when TaskRunner instantiated.

Reply:
For this I believe there's a similar problem: if users change this flag on the fly, the path has to be recreated. If we create the path when TaskRunner is instantiated, there's no way to recreate the path.

Reply:
Same reason as the previous, dynamic update is overkill.

Reply:
Concluded to use one local variable when necessary.